tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ScalaJSClassEmitter.scala

```diff
@@ -64,6 +66,40 @@ private[scalajs] final class ScalaJSClassEmitter(
     linkedClassByName(className).kind == ClassKind.Interface
   }
 
+  private[emitter] lazy val candidateForJSConstructorOpt: Set[String] = {
+    /* Those classes appear in hard coded js files.
+     * Their constructors aren't always present in the desugared code.
+     * As such we cannot optimize those as we can't change the hard coded files.
+     */
+    val blackList = Set("jl_ClassCastException",
+      "sjsr_UndefinedBehaviorError", "jl_NullPointerException",
+      "js_CloneNotSupportedException")
+
+    val instantiatedClasses = linkingUnit.classDefs.filter(_.hasInstances)
+    val cantCtorOpt =
+      (instantiatedClasses.flatMap(_.superClass).map(_.name).toSet ++ blackList)
+
+    (for {
+      classDef <- linkingUnit.classDefs
+      if !cantCtorOpt(classDef.encodedName)
+      if classDef.memberMethods.count(
+          x => isConstructorName(x.info.encodedName)) == 1
+      if classDef.classExports.isEmpty
+    } yield {
+      classDef.encodedName
+    }).toSet
+  }
+
+  private[emitter] def usesJSConstructorOpt(className: String) =
+    incClassEmitter.usesJSConstructorOpt(className)
```

Comment:
As previously commented on a similar issue, I no longer use it anywhere else than this class.